sw/device/tests/entropy_src_bypass_mode_health_test.c

```diff
@@ -0,0 +1,563 @@
+  // Using entropy test utility to stop all EDN0, EDN1, CSRNG, and the Entropy
+  // Source
+  LOG_INFO("Disabling the entropy complex...");
+  TRY(entropy_testutils_stop_all());
+  return OK_STATUS();
+}
+
+status_t configure_loose_health_tests(void) {
+  LOG_INFO("Configuring loose health test thresholds...");
+
+  // Check if entropy source is locked
+  bool is_locked;
+  TRY(dif_entropy_src_is_locked(&entropy_src, &is_locked));
+  if (is_locked) {
+    LOG_ERROR("Entropy source is locked. Cannot configure health tests");
+    return INTERNAL();
+  }
+
+  // Configure Repetition Count Test
+  dif_entropy_src_health_test_config_t repcnt_test_config = {
+      .test_type = kDifEntropySrcTestRepetitionCount,
+      .high_threshold = 512,
+      .low_threshold = 0,
+  };
+  TRY(dif_entropy_src_health_test_configure(&entropy_src, repcnt_test_config));
+
+  // Configure Adaptive Proportion Test
+  dif_entropy_src_health_test_config_t adaptp_test_config = {
+      .test_type = kDifEntropySrcTestAdaptiveProportion,
+      .high_threshold = 512,
+      .low_threshold = 0,
+  };
+  TRY(dif_entropy_src_health_test_configure(&entropy_src, adaptp_test_config));
+
+  // Configure Bucket Test
+  dif_entropy_src_health_test_config_t bucket_test_config = {
+      .test_type = kDifEntropySrcTestBucket,
+      .high_threshold = 512,
+      .low_threshold = 0,
+  };
+  TRY(dif_entropy_src_health_test_configure(&entropy_src, bucket_test_config));
+
+  // Configure Markov Test
+  dif_entropy_src_health_test_config_t markov_test_config = {
+      .test_type = kDifEntropySrcTestMarkov,
+      .high_threshold = 512,
+      .low_threshold = 0,
+  };
+  TRY(dif_entropy_src_health_test_configure(&entropy_src, markov_test_config));
+
+  return OK_STATUS();
+}
+
+status_t enable_loose_entropy_src_bypass_mode(void) {
+  LOG_INFO("Enabling ENTROPY_SRC in bypass mode...");
+
+  // Ensure the entropy source is not locked
+  bool is_locked;
+  TRY(dif_entropy_src_is_locked(&entropy_src, &is_locked));
+  if (is_locked) {
+    LOG_ERROR("Entropy source is locked. Cannot configure ENTROPY_SRC");
+    return INTERNAL();
+  }
+
+  // Configure ENTROPY_SRC in bypass mode with health tests enabled
+  dif_entropy_src_config_t config = {
+      .fips_enable = false,
+      .route_to_firmware = false,
+      .fips_flag = false,
+      .rng_fips = false,
+      .bypass_conditioner = true,
+      .single_bit_mode = kDifEntropySrcSingleBitModeDisabled,
+      .health_test_threshold_scope = false,
+      .health_test_window_size = 512,
+      .alert_threshold = 0xFFFF,
+  };
+
+  // Apply the configuration and enable ENTROPY_SRC
+  dif_result_t result =
+      dif_entropy_src_configure(&entropy_src, config, kDifToggleEnabled);
+  // DEBUG
+  if (result != kDifOk) {
+    LOG_ERROR("dif_entropy_src_configure failed: %d", result);
+    return INTERNAL();
+  }
+
+  uint32_t errors;
+  TRY(dif_entropy_src_get_errors(&entropy_src, &errors));
+  LOG_INFO("ENTROPY_SRC Errors: 0x%x", errors);
```

Comment:
What is the expected result here? We should add a check whether we get the expected result.
Same applies throughout the test.

Reply:
These were used for debugging if there hang is due to some other cases
I shall remove them